### PR TITLE
Replace CV_USRTYPE1 for int64 to CV_32SC2 in Torch importer

### PR DESCRIPTION
**Pull request by dkurt**
```
allow_multiple_commits=1
```

### CODE REVIEW

File: modules/dnn/src/torch/torch_importer.cpp [hunk 1]
```diff
@@ -257,7 +257,7 @@ struct TorchImporter
         case CV_32S:
             THFile_readIntRaw(file, (int*)storageMat.data, size);
             break;
-        case CV_USRTYPE1:
+        case CV_32SC2:
         {
             double *buf = storageMat.ptr<double>();
```
Comment (alalek) [inline modules/dnn/src/torch/torch_importer.cpp]: > `.ptr<double>()`

```
int64_t *buf = storageMat.ptr<int64_t>();
```

File: modules/dnn/src/torch/torch_importer.cpp [hunk 2]
```diff
@@ -236,7 +236,7 @@ struct TorchImporter
     void readTorchStorage(int index, int type = -1)
     {
         long size = readLong();
-        Mat storageMat(1, size, (type != CV_USRTYPE1) ? type : CV_64F); //handle LongStorage as CV_64F Mat
+        Mat storageMat(1, size, (type != CV_32SC2) ? type : CV_64F); //handle LongStorage as CV_64F Mat
```
Comment (alalek) [inline modules/dnn/src/torch/torch_importer.cpp]: Check is not required here.

File: modules/dnn/src/torch/torch_importer.cpp (hunk 1 above)
Comment (dkurt) [reply modules/dnn/src/torch/torch_importer.cpp]: @alalek, The thing is that it converts int64 data to doubles. I modified parser so now we use enum variables instead OpenCV's types.

File: modules/dnn/src/torch/torch_importer.cpp [hunk 3]
```diff
@@ -236,36 +248,44 @@ struct TorchImporter
...
-        case CV_32S:
+        case TYPE_INT:
+            storageMat.create(1, size, CV_32S);
             THFile_readIntRaw(file, (int*)storageMat.data, size);
             break;
-        case CV_USRTYPE1:
+        case TYPE_LONG:
         {
+            storageMat.create(1, size, CV_64F);   //handle LongStorage as CV_64F Mat
             double *buf = storageMat.ptr<double>();
```
Comment (alalek) [inline modules/dnn/src/torch/torch_importer.cpp]: BTW, `.ptr<T>()` doesn't make type checks about `cv::Mat` elements like `.at<T>()` method do (in Debug mode).

**Pull request merged**